forms/gridfield/GridFieldSortableHeader.php

```diff
@@ -233,9 +233,16 @@ public function getManipulatedData(GridField $gridField, SS_List $dataList) {
 					// been aliased. In that event, assume what the user has
 					// provided is the correct value
 					if(!$joinClass) $joinClass = $lastAlias;
-
+					
+					// Check on which table is the sort column
+					$table = ClassInfo::table_for_object_field($tmpItem->class, $parts[sizeof($parts)-1]);
```

Comment:
Shouldn't this be $methodName? Also, the field will be `$methodName . 'ID'` for has_one relations.

You are using $parts[sizeof($parts)-1] within a loop, which will always join on the last segment in the dot separated column list, even when joining earlier tables.